src/transformers/quantizers/quantizer_quanto.py

```diff
@@ -0,0 +1,211 @@
+
+
+if TYPE_CHECKING:
+    from ..modeling_utils import PreTrainedModel
+
+from ..utils import is_quanto_available, is_torch_available, logging
+from ..utils.quantization_config import QuantoConfig
+
+
+if is_torch_available():
+    import torch
+
+logger = logging.get_logger(__name__)
+
+
+class QuantoHfQuantizer(HfQuantizer):
+    """
+    Quantizer for the quanto library
+    """
+
+    required_packages = ["quanto", "accelerate"]
+    requires_parameters_quantization = True
+    requires_calibration = False
+
+    def __init__(self, quantization_config: QuantoConfig, **kwargs):
+        super().__init__(quantization_config, **kwargs)
+        self.post_init()
+
+    def post_init(self):
+        r"""
+        Safety checker
+        """
+        if self.quantization_config.activations is not None and not self.pre_quantized:
+            raise ValueError(
+                "We don't support quantizing the activations with transformers library."
+                "Use quanto library for more complex use cases such as activations quantization, calibration and quantization aware training."
+            )
+
+    def validate_environment(self, *args, **kwargs):
+        if not is_quanto_available():
+            raise ImportError("Loading a quanto quantized model requires quanto library (`pip install quanto`)")
+        device_map = kwargs.get("device_map", None)
+        if device_map is not None and isinstance(device_map, dict):
+            if "cpu" in device_map.values() or "disk" in device_map.values():
+                if version.parse(importlib.metadata.version("accelerate")) <= version.parse("0.27.0"):
+                    raise ValueError(
+                        "You have a version of `accelerate` that is not compatible cpu/disk offload with quanto quantized model. "
+                        "You need to install a version of accelerate > 0.27.0."
+                    )
+
+    def update_torch_dtype(self, torch_dtype: "torch.dtype") -> "torch.dtype":
+        if torch_dtype is None:
+            # TODO: Discuss if we should do that for quanto. I think that we should probably not do that and let the user cast the torch_dtype by themselves.
+            # since in this case, quanto can also work on cpu.
+            # If a user have both a cpu and cuda and he wants to play with quanto on cpu, he will have a specify manually torch_dtype to torch.float32.
+            if torch.cuda.is_available():
+                torch_dtype = torch.float16
+                logger.info(
+                    "CUDA available. Assuming Quanto inference on GPU and loading the model in `torch.float16`. To overwrite it, set `torch_dtype` manually."
+                )
+            else:
+                torch_dtype = torch.float32
+                logger.info(
+                    "CUDA is unavailable. Assuming AQLM inference on CPU and loading the model in `torch.float32`. To overwrite it, set `torch_dtype` manually."
+                )
```

Comment:
```suggestion
```
I suggest we don't do anything here because quanto is quite agnostic to CPU / GPU devices, what do you think?